NMS-13472: update table formatting in docs (#3562)
OpenNMS/opennms: docs/modules/operation/pages/database-reports/database.adoc

To receive data from the Measurements API, create a query as follows:

.Sample queryString to receive data from the Measurements API
[source, xml]
------
<query-request step="300000" start="$P{startDateTime}" end="$P{endDateTime}" maxrows="2000"> <1>
  <source aggregation="AVERAGE" label="IfInOctets" attribute="ifHCInOctets" transient="false" resourceId="node[$P\{nodeid}].interfaceSnmp[$P{interface}]"/>
  <source aggregation="AVERAGE" label="IfOutOctets" attribute="ifHCOutOctets" transient="false" resourceId="node[$P\{nodeid}].interfaceSnmp[$P{interface}]"/>
</query-request>
------
<1> The query language. In our case, "measurement", but JasperReports supports a lot out of the box, such as sql, xpath, and so on.

[[ga-database-reports-fields]]
=== Fields

Each data source should return a number of fields, which you can use in the report.
The Measurement Datasource supports the following fields:

[options="header"]
[cols="1,1,3"]
|===
| Field name  | Field type         | Field description

> | <label>   | java.lang.Double | Each source defined as `transient=false` can be used as a field.

Reviewer: Should this have the < > surrounding the name?

Author: nope. thanks!

                                                                                                                              `snmphyaddr`: String, the MAC address of the interface; for example, `005e607e9e00`. May be null. +
                                                                                                                              As each input parameter may be null, not all of them can be null at the same time. At least one input parameter has to be defined. +
                                                                                                                              +
                                                                                                                              For more details, see <<ga-database-reports-interface-descriptor, Interface descriptor use>>.
|===

[[ga-database-reports-node-source-descriptor]]
=== Node source descriptor use

A node is addressed by a node source descriptor.
The node source descriptor references the node either via the foreign source and foreign id or by the node id.

> If `store by foreign source` is enabled, is is possible to address the node only via `foreign source` and `foreign id`.

Reviewer: ```suggestion
If `storeByForeignSource` is enabled, it is only possible to address the node via foreign source and foreign id.
```

To make report creation easier, there is a helper method to create the node source descriptor.

> NOTE: For more information about store by foreign source, see link:https://wiki.opennms.org/wiki/ForeignSource/foreignId_Data_Storage_How-To[our Wiki].

Reviewer: Should not have wiki links - and the referenced wiki page has also been moved to https://opennms.discourse.group/t/storing-data-with-foreign-sources/2057, though the content of this is something that should exist somewhere in the docs as a best practice.

Author: Updated and created Jira: https://issues.opennms.org/browse/NMS-13482

                                                                                                                              +
                                                                                                                              For more details, see <<ga-database-reports-interface-descriptor, Interface descriptor use>>.
|===

[[ga-database-reports-node-source-descriptor]]
=== Node source descriptor use

A node is addressed by a node source descriptor.
The node source descriptor references the node either via the foreign source and foreign id or by the node id.

If `storeByForeignSource` is enabled, it is only possible to address the node via foreign source and foreign id.

To make report creation easier, there is a helper method to create the node source descriptor.

NOTE: For more information, see https://opennms.discourse.group/t/storing-data-with-foreign-sources/2057[Storing data with foreign sources] on Discourse.

The following example shows the use of that helper.

.jrxml report snippet to visualize the use of the node source descriptor.
[source, xml]
------
<parameter name="nodeResourceDescriptor" class="java.lang.String" isForPrompting="false">

>   <defaultValueExpression><![CDATA[org.opennms.netmgt.jasper.helper.MeasurementsHelper.getNodeOrNodeSourceDescriptor(String.valueOf($Pnodeid), $Pforeignsource, $Pforeignid)]]></defaultValueExpression>

Reviewer: ```suggestion
  <defaultValueExpression><![CDATA[org.opennms.netmgt.jasper.helper.MeasurementsHelper.getNodeOrNodeSourceDescriptor(String.valueOf($P{nodeid}), $Pforeignsource, $P{foreignid})]]></defaultValueExpression>
```

<source aggregation="AVERAGE" label="IfOutOctets" attribute="ifHCOutOctets" transient="false" resourceId="$P{nodeResourceDescriptor}.interfaceSnmp[en0-005e607e9e00]"/>
</query-request>]]>
------
Depending on the input parameters, you get either a node resource descriptor or a foreign source/foreign id resource descriptor.

[[ga-database-reports-interface-descriptor]]
=== Interface descriptor use

An interfaceSnmp is addressed with the exact interface descriptor.
To allow easy access to the interface descriptor, we provide a helper tool.
The following example shows the use of that helper.

> .jrxml report snippet to visualize the use of the interface descriptor

Reviewer: There are two `\` in the following source block that can be removed.

Author: I see 5. Can I remove all of them?

</parameter>
<queryString language="Measurement">
  <![CDATA[<query-request step="300000" start="$P{startDateTime}" end="$P{endDateTime}" maxrows="2000">

> <source aggregation="AVERAGE" label="IfInOctets" attribute="ifHCInOctets" transient="false" resourceId="node[$Pnodeid].interfaceSnmp[$P{interface}]"/>

Reviewer: ```suggestion
<source aggregation="AVERAGE" label="IfInOctets" attribute="ifHCInOctets" transient="false" resourceId="node[$P{nodeid}].interfaceSnmp[$P{interface}]"/>
```

> <source aggregation="AVERAGE" label="IfOutOctets" attribute="ifHCOutOctets" transient="false" resourceId="node[$Pnodeid].interfaceSnmp[$P{interface}]"/>

Reviewer: ```suggestion
<source aggregation="AVERAGE" label="IfOutOctets" attribute="ifHCOutOctets" transient="false" resourceId="node[$P{nodeid}].interfaceSnmp[$P{interface}]"/>
```
